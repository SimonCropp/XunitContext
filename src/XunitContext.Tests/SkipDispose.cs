using System;
using Xunit;
using Xunit.Abstractions;

public class SkipDispose :
    XunitContextBase
{
    [Fact]
    public void Dispose_should_flush()
    {
        Write("part1");
        Write(" part2");
        base.Dispose();
        ObjectApprover.Verify(Logs);
    }

    [Fact]
    public void Write_after_dispose_should_throw()
    {
        base.Dispose();
        var exception = Assert.Throws<Exception>(WriteLine);
        ObjectApprover.Verify(exception);
    }

    public SkipDispose(ITestOutputHelper testOutput) :
        base(testOutput)
    {
    }

    public override void Dispose()
    {
    }
}