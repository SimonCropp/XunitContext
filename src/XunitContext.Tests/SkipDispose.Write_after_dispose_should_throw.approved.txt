{
  Message: 'Logging context has been flushed. Could not write the string: Environment.NewLine',
  Data: {},
  Source: 'XunitLogger'
}